e2etest/pkg/status/l2status.go

```diff
@@ -4,37 +4,29 @@ package status
 
 import (
 	"context"
-	"fmt"
+	"go.universe.tf/e2etest/pkg/metallb"
 
 	"go.universe.tf/metallb/api/v1beta1"
 	v1 "k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/api/errors"
-	"k8s.io/apimachinery/pkg/types"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-func GetL2Status(cs client.Client, svc *v1.Service, nodeName string) (*v1beta1.ServiceL2Status, error) {
-	s := &v1beta1.ServiceL2Status{}
-	if err := cs.Get(context.TODO(), types.NamespacedName{
-		Namespace: svc.Namespace,
-		Name:      fmt.Sprintf("%s-%s", svc.Name, nodeName),
-	}, s); err != nil {
-		return nil, err
-	}
-	return s, nil
-}
+const (
+	LabelServiceName      = "metallb.io/service-name"
+	LabelServiceNamespace = "metallb.io/service-namespace"
+)
 
-func GetSvcPossibleL2Status(cs client.Client, svc *v1.Service, nodes *v1.NodeList) ([]*v1beta1.ServiceL2Status, error) {
-	var l2Statuses []*v1beta1.ServiceL2Status
-	for _, node := range nodes.Items {
-		s, err := GetL2Status(cs, svc, node.Name)
-		if err != nil && errors.IsNotFound(err) {
-			continue
-		}
-		if err != nil {
-			return nil, err
-		}
-		l2Statuses = append(l2Statuses, s)
+func L2ForService(cs client.Client, svc *v1.Service) ([]v1beta1.ServiceL2Status, error) {
+	statusList := v1beta1.ServiceL2StatusList{}
```

Comment:
if we return error here if we get more than one instance, the calling sites will be cleaner (also, it feels weird to read "status.ForService" and then see that we are receiving more than one status.